### DISCUSSION TITLE
Access to undeclared static property Spatie\Permission\PermissionRegistrar::$pivotPermission

### DISCUSSION THREAD

**Discussion by SubhamKumar101201** [spatie/laravel-permission#2576] [Q&A] [answered]
```
  2023_10_25_103252_create_permission_tables ............................................................................................. 43ms FAIL

   Error 

  Access to undeclared static property Spatie\Permission\PermissionRegistrar::$pivotPermission

  at database/migrations/2023_10_25_103252_create_permission_tables.php:54
     50▕             }
     51▕         });
     52▕ 
     53▕         Schema::create($tableNames['model_has_permissions'], function (Blueprint $table) use ($tableNames, $columnNames, $teams) {
  ➜  54▕             $table->unsignedBigInteger(PermissionRegistrar::$pivotPermission);
     55▕ 
     56▕             $table->string('model_type');
     57▕             $table->unsignedBigInteger($columnNames['model_morph_key']);
     58▕             $table->index([$columnNames['model_morph_key'], 'model_type'], 'model_has_permissions_model_id_model_type_index');

      +4 vendor frames 

  5   database/migrations/2023_10_25_103252_create_permission_tables.php:53
      Illuminate\Support\Facades\Facade::__callStatic()
      +36 vendor frames 

  42  artisan:35
      Illuminate\Foundation\Console\Kernel::handle()

```

**Comment by SubhamKumar101201** [comment]
"althinect/filament-spatie-roles-permissions": "^2.1",(v2.2.8) after update, 
"althinect/filament-spatie-roles-permissions": "^2.1", (v2.1.1) before update

**Comment by erikn69** [comment, marked as answer]
https://github.com/spatie/laravel-permission/blob/main/docs/upgrading.md#upgrade-essentials

**Reply by Samimi-Information-Technology** [reply]
This did not work for me.

The proper solution that worked for me, and to this issue is by bokele linked below:

https://github.com/spatie/laravel-permission/discussions/2576#discussioncomment-8417545

**Comment by cristian963** [comment]
Whay did you do? I have the same problem.

**Reply by drbyte** [reply]
Update your migration, as described in the upgrade documentation.

**Comment by bokele** [comment]
The easiest way is to run :

1. composer update
2. Delete the permission migration
3. php artisan vendor:publish --provider="Spatie\Permission\PermissionServiceProvider"
4. php artisan migrate

**Reply by CaptainX2019** [reply]
This is the great solution worked fine for me thanx dude

**Reply by CeruttiMaicon** [reply]
Thank you, it helped me a lot, I worked here for my update to Laravel 11

**Reply by erikn69** [reply]
It is correct as long as you have not made any customization in the migration file

**Reply by webblufter** [reply]
you saved time

**Reply by Samimi-Information-Technology** [reply]
It worked!

Thank you.

**Reply by KeyzerT** [reply]
You just saved us. It worked thank you

**Reply by Sparrowan** [reply]
> 2\. migration

This worked. Thanks.

**Reply by HoangMen1902** [reply]
Thanks for the solution <3

**Reply by PaulChinks** [reply]
thanks this worked for me

**Reply by fabiorodriguesroque** [reply]
Bad solution, if you are getting this problem you should follow the instructions here: [Upgrade Essentials](https://github.com/spatie/laravel-permission/blob/main/docs/upgrading.md#upgrade-essentials)

If you are struggling to find the new migration file, it's here: [create_permission_tables](https://github.com/spatie/laravel-permission/blob/main/database/migrations/create_permission_tables.php.stub)

**Reply by DarAzizi** [reply]
> 4\. php artisan migrate

Best way .. thank you

**Comment by OleksandrMakar01** [comment]
[erikn69] of answer is wrong.

**Comment by doanthean** [comment]
~Delete all content from file xxx_xxx_xxx_create_permission_table.php, paste the new migration from this package.~

EDIT: Follow the upgrade instructions and re-publish the migration for the new version.
https://github.com/spatie/laravel-permission/blob/main/docs/upgrading.md#upgrade-essentials

**Reply by drbyte** [reply]
Better to just follow the upgrade instructions instead of pasting someone else's code without inspection:
https://github.com/spatie/laravel-permission/blob/main/docs/upgrading.md#upgrade-essentials

**Reply by ParagiDev** [reply]
Thank you, it works 👍
